### PR TITLE
Students

### PR DESCRIPTION
**Pull request by jackcarlisle**
# ready for review

- [x] adds the functionality for a teacher to create students within a group 
- [x] adds tests

#27

### CODE REVIEW

File: test/controllers/student_controller_test.exs [hunk 1]
```diff
@@ -0,0 +1,144 @@
+defmodule Skillswheel.StudentControllerTest do
+  use Skillswheel.ConnCase
+  alias Skillswheel.Group
+  alias Skillswheel.Student
+  alias Skillswheel.User
+  alias Skillswheel.UserGroup
```
Comment (samhstn) [inline test/controllers/student_controller_test.exs]: `alias Skillswheel.{ConnCase, Group, Student, User, UserGroup}` ?

Let me know if your keen to use this pattern, I like it but I'm fine with not implementing if your not a fan

File: test/controllers/student_controller_test.exs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    %Student{
+      id: 4,
+      first_name: "First",
+      last_name: "Last",
+      sex: "male",
+      year_group: "2",
+      group_id: 4
+    } |> Repo.insert
+
```
Comment (samhstn) [inline test/controllers/student_controller_test.exs:116]: I'm finding these inserts a little long to read, could we extract out what is being done into a helper function in `test/support/test_helpers.ex`

**Review by samhstn — CHANGES_REQUESTED** [review]
Amazing progress, glad to finally be over that many to many relationship, plain sailing from now I guess? 👍 

Let me know your thoughts on the couple of points raised, then it's good to go.

Also I feel that perhaps we could look to employ [this](https://github.com/dwyl/learn-phoenix-framework/pull/42/files#diff-43818413585772311e7de1350bfc0542R14) pattern in places in the tests

File: test/controllers/student_controller_test.exs (hunk 1 above)
Comment (jackcarlisle) [reply test/controllers/student_controller_test.exs]: Sounds good I'll swap that over!

File: test/controllers/student_controller_test.exs (hunk 2 above)
Comment (jackcarlisle) [reply test/controllers/student_controller_test.exs:116]: Yeah sounds good, I'll look into this tomorrow

**Pull request merged**